client/core/bookie.go

```diff
@@ -244,12 +246,17 @@ func (c *Core) Book(dex string, base, quote uint32) (*OrderBook, error) {
 	dc.booksMtx.RLock()
 	book, found := dc.books[mkt]
 	dc.booksMtx.RUnlock()
+	// If not found, attempt to make a temporary subscription and return the
+	// initial book.
 	if !found {
-		return nil, fmt.Errorf("no market %s", mkt)
+		book, bookFeed, err := c.Sync(dex, base, quote)
+		if err != nil {
+			return nil, fmt.Errorf("unable to sync book: %v", err)
+		}
+		bookFeed.Close()
+		return book, nil
```

Comment:
This changes the behavior of this function quite a bit, but I cannot see that it has caused any problems. Initially I wasn't sure how to fetch a one-time book, and I was considering more costly options, but if there is already a subscription to the book, this is very cheap. Otherwise making a subscription and canceling is the only way I can see to get market orders unless another route is added to the server.

Reply:
That's pretty much the way for now, although we've been planning on a http api for this kind of thing.